src/lib/containers/Goals.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react'
+import {
+  QueryResultBundle,
+  QueryBundleRequest,
+  FileHandleAssociation,
+  FileHandleAssociateType,
+  BatchFileRequest,
+} from '../utils/synapseTypes'
+import { SynapseClient, SynapseConstants } from '../utils'
+import { SynapseClientError, getFiles } from '../utils/SynapseClient'
+import { Error } from '../containers/Error'
+import QueryCount from './QueryCount'
+import { parseEntityIdFromSqlStatement } from '../utils/functions/sqlFunctions'
+
+export type GoalsProps = {
+  sql: string
```

Comment:
Another approach would be to just take the tableID in this component, and define the columns explicitly in the (sql baked into this component).  Then you wouldn't need to do any client-side validation, since the query execution would fail if any of the expected columns were missing.  `select TableId, Title, Summary, Link, Asset from ${entityId} order by ItemOrder`

Reply:
ah, I like it when synapse does the work! I will change it to that